src/Network/IPv4Network.php

```diff
@@ -42,10 +43,16 @@ public function subnet(): SubnetInterface
         return $this->subnet;
     }
 
-    public function contains(AddressInterface $address): bool
+    public function contains(AddressableInterface $address): bool
     {
-        return (strcmp($address->inAddr(), $this->getFirstIP()->inAddr()) >= 0)
-            && (strcmp($address->inAddr(), $this->getLastIP()->inAddr()) <= 0);
+        if ($address->subnet() !== null
+            && !$this->subnet->contains($address->subnet())
+        ) {
+            return false;
+        }
```

Comment:
I don't believe subnet needs to be nullable and with that the `$this->subnet->contains($address->subnet())` can be added to the end of the return statement. Which is nicer I think, it is also more logical to first check the address and after that the subnet, eventhough the result is the same it reads more logical

Reply:
Yea consider it done.